vision_datasets/image_object_detection/detection_as_kvp_dataset.py

```diff
@@ -44,24 +61,51 @@ def __init__(self, detection_dataset: VisionDataset):
 
             kvp_label_data = self.construct_kvp_label_data(bboxes)
             img_ids = [self.img_id_to_pos[img.id]]  # 0-based index
-            kvp_annotation = KeyValuePairLabelManifest(id, img_ids, label_data=kvp_label_data)
+            kvp_annotation = KeyValuePairLabelManifest(
+                id, img_ids, label_data=kvp_label_data
+            )
 
             # KVPDatasetManifest expects img.labels to be empty. Labels are instead stored in KVP annotation
             img.labels = []
             annotations.append(kvp_annotation)
 
-        dataset_manifest = KeyValuePairDatasetManifest(detection_dataset.dataset_manifest.images, annotations, schema, additional_info=detection_dataset.dataset_manifest.additional_info)
-        super().__init__(dataset_info, dataset_manifest, dataset_resources=detection_dataset.dataset_resources)
-
-    def construct_schema(self, class_names: List[str]) -> Dict[str, Any]:
+        dataset_manifest = KeyValuePairDatasetManifest(
+            detection_dataset.dataset_manifest.images,
+            annotations,
+            schema,
+            additional_info=detection_dataset.dataset_manifest.additional_info,
+        )
+        super().__init__(
+            dataset_info,
+            dataset_manifest,
+            dataset_resources=detection_dataset.dataset_resources,
+        )
+
+    def construct_schema(
+        self, include_class_names: bool, custom_schema_description: Union[str, None]
+    ) -> Dict[str, Any]:
         schema: Dict[str, Any] = self.DETECTION_SCHEMA  # initialize with base schema
-        schema["fieldSchema"][f"{self.OBJECTS_KEY}"]["items"]["classes"] = {c: {"description": f"Always output {c} as the class."} if len(class_names) == 1 else {} for c in class_names}
+        if include_class_names:
+            schema["fieldSchema"][f"{self.OBJECTS_KEY}"]["items"]["classes"] = {
+                c: {"description": f"Always output {c} as the class."}
+                if len(self.class_names) == 1
+                else {}
+                for c in self.class_names
+            }
+        else:
+            del schema["fieldSchema"][f"{self.OBJECTS_KEY}"]["items"]["classes"]
+
+        if custom_schema_description is not None:
+            schema["description"] = custom_schema_description
+
         return schema
 
     def construct_kvp_label_data(self, bboxes: List[List[int]]):
         raise NotImplementedError
```

Comment:
should we `raise NotImplementedError()` to make sure we are raise the exception object, not the type object?

Reply:
updated